plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codemodernizer/model/CodeModernizerArtifact.kt

```diff
@@ -99,6 +101,14 @@ open class CodeModernizerArtifact(
             if (!patchesDir.isDirectory()) {
                 throw RuntimeException("Expected root for patches was not a directory.")
             }
+            val patch = patchesDir.walk().map { fileSystem.findFileByNioFile(it) }.toList()
+            if (patch[0] == null) {
+                ApplicationManager.getApplication().invokeLater {
```

Comment:
Will this guarantee `syncRefresh` to run before the next `patchesDir.walk()` call? Not super familiar with invokeLater

Reply:
Question: I'm not sure I understand the problem here. Calling `invokeLater` with the sync needs to happen before the `return patchesDir.walk()` statement correct?

This `invokeLater` function is async so the the `return patchesDir.walk()` will possibly get called before the `invokeLater` finishes.

Reply:
Check this example output:
```
    println("Main thread: Before invokeLater")

    SwingUtilities.invokeLater {
        println("Running on EDT")
        Thread.sleep(1500) // Simulate a long-running task
        println("Done running on EDT")
    }


    println("Main thread: After invokeLater")
    Thread.sleep(2000) // Wait for 1 second
    println("Main thread: Exiting")
```

Outputs the following:

```
Main thread: Before invokeLater
Main thread: After invokeLater
Running on EDT
Done running on EDT
Main thread: Exiting
```

Reply:
If you want to test something out quickly. Try the online [kotlin playground](https://play.kotlinlang.org/#eyJ2ZXJzaW9uIjoiMS45LjIyIiwicGxhdGZvcm0iOiJqYXZhIiwiYXJncyI6IiIsIm5vbmVNYXJrZXJzIjp0cnVlLCJ0aGVtZSI6ImlkZWEiLCJjb2RlIjoiaW1wb3J0IGphdmF4LnN3aW5nLlN3aW5nVXRpbGl0aWVzXG4vKipcbiAqIFlvdSBjYW4gZWRpdCwgcnVuLCBhbmQgc2hhcmUgdGhpcyBjb2RlLlxuICogcGxheS5rb3RsaW5sYW5nLm9yZ1xuICovXG5mdW4gbWFpbigpIHtcblxuICAgIHByaW50bG4oXCJNYWluIHRocmVhZDogQmVmb3JlIGludm9rZUxhdGVyXCIpXG5cblxuICAgIFN3aW5nVXRpbGl0aWVzLmludm9rZUxhdGVyIHtcblxuICAgICAgICBwcmludGxuKFwiUnVubmluZyBvbiBFRFRcIilcblxuICAgICAgICBUaHJlYWQuc2xlZXAoMTUwMCkgLy8gU2ltdWxhdGUgYSBsb25nLXJ1bm5pbmcgdGFza1xuXG4gICAgICAgIHByaW50bG4oXCJEb25lIHJ1bm5pbmcgb24gRURUXCIpXG5cbiAgICB9XG5cblxuICAgIHByaW50bG4oXCJNYWluIHRocmVhZDogQWZ0ZXIgaW52b2tlTGF0ZXJcIilcblxuICAgIFRocmVhZC5zbGVlcCgyMDAwKSAvLyBXYWl0IGZvciAxIHNlY29uZFxuXG5cbiAgICBwcmludGxuKFwiTWFpbiB0aHJlYWQ6IEV4aXRpbmdcIilcbn0ifQ==)

Reply:
That's a good point. We also have `invokeAndWait` which appears to work fine too. Though it seems like you're not supposed to call that from the EDT as it will lead to deadlock and throw an Exception. This didn't happen for me but want to learn more about this before doing it. Also want @arangatang to give his thoughts.

Reply:
what about `refreshAndFindFileByPath`? no need to refresh the entire VFS